### PR TITLE
Feat/url to base64

### PR DESCRIPTION
**Pull request by Surendhar1997**
worked on the changes to convert the given url to base64 with/without metadata.

### CODE REVIEW

File: core/src/main/java/com/fincity/saas/core/functions/file/UrlToBase64.java
Comment (serialcoder13) [inline core/src/main/java/com/fincity/saas/core/functions/file/UrlToBase64.java:1]: Change the function name to FileToBase64

File: core/src/main/java/com/fincity/saas/core/feign/IFeignFilesService.java [hunk 1]
```diff
@@ -36,4 +36,12 @@ Mono<ByteBuffer> downloadFile(
                         @RequestParam(required = false, defaultValue = "HORIZONTAL") String resizeDirection,
                         @RequestParam(required = false, defaultValue = "false") Boolean noCache,
                         @RequestParam(required = false) String name);
+
+
+        @GetMapping("/api/files/internal/{resourceType}/convertToBase64")
+        Mono<String> convertToBase64(
```
Comment (serialcoder13) [inline core/src/main/java/com/fincity/saas/core/feign/IFeignFilesService.java:42]: readFileAsBase64

File: files/src/main/java/com/fincity/saas/files/service/AbstractFilesResourceService.java [hunk 2]
```diff
@@ -866,6 +867,78 @@ public Mono<FileDetail> createInternal(String clientCode, boolean override, Stri
 
 	}
 
+	public Mono<String> convertToBase64(String clientCode, String resourceType, String uri, boolean metadataRequired) {
```
Comment (serialcoder13) [inline files/src/main/java/com/fincity/saas/files/service/AbstractFilesResourceService.java:870]: readFileAsBase64

File: files/src/main/java/com/fincity/saas/files/service/FilesAccessPathService.java [hunk 3]
```diff
@@ -383,4 +383,12 @@ public Mono<FilesAccessPath> createInternalAccessPath(FilesAccessPath accessPath
 
 		return super.create(accessPath);
 	}
+
+	public Mono<Boolean> isClientBeingManaged(String managingClientCode, String clientCode) {
+
+		if(StringUtil.safeEquals(managingClientCode, clientCode) || StringUtil.safeEquals(managingClientCode, "SYSTEM"))
+			return Mono.just(true);
+			
+		return this.securityService.isBeingManaged(managingClientCode, clientCode);
```
Comment (serialcoder13) [inline files/src/main/java/com/fincity/saas/files/service/FilesAccessPathService.java:392]: This method take care of SYSTEM check. Don't check here.

**Review by serialcoder13 — CHANGES_REQUESTED** [review]
Please make the changes

File: core/src/main/java/com/fincity/saas/core/functions/file/UrlToBase64.java
Comment (Surendhar1997) [reply core/src/main/java/com/fincity/saas/core/functions/file/UrlToBase64.java:1]: done

**Review by serialcoder13 — APPROVED** [review]
LGTM

**Pull request merged**